docs/source/assets_writer.rst

```diff
@@ -0,0 +1,103 @@
+.. Copyright 2016 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+.. cartographer SHA: 30f7de1a325d6604c780f2f74d9a345ec369d12d
+.. cartographer_ros SHA: 44459e18102305745c56f92549b87d8e91f434fe
+
+
+Assets writer
+=============
+
+The purpose of SLAM is to compute the trajectory of a single sensor through a metric space.
+On a higher level, the input of SLAM is sensor data, its output is the best estimate of the trajectory up to this point in time.
+To be real-time and efficient, Cartographer throws most of the sensor data away immediately.
+
+The trajectory alone is rarely of interest.
+But once the best trajectory is established, the full sensor data can be used to derive and visualize information about its surroundings.
+
+Cartographer provides the assets writer for this.
+Its input is the Cartographer state, the sensor data originally fed to SLAM, the sensor extrinsics and a pipeline configuration file.
```

Comment:
The inputs could be clearer, as it'd not obvious to someone new to Cartographer that the "state" refers to the .pbstream output by the SLAM algorithm (i.e. by running demo_backpack_3d.launch). It's also not clear what "sensor intrinsics" means - is this referring to data in the .bag or .pbstream files? Or is it referring to separate file included by the asset writer?

Bullet points might be clearer here, e.g.:

There are three inputs to the asset writer:
1. The original sensor data fed to SLAM in a rosbag file
2. The cartographer state, which is contained in the .pbstream file that SLAM creates
3. A pipeline configuration, which is defined in a .lua file